tests/vsock.rs

```diff
@@ -71,3 +76,25 @@ fn test_vsock() {
 fn test_get_local_cid() {
     assert_eq!(get_local_cid().unwrap(), VMADDR_CID_HOST);
 }
+
+#[test]
+fn test_listener_local_addr() {
+    let listener = VsockListener::bind(&VsockAddr::new(VMADDR_CID_HOST, LISTEN_PORT)).unwrap();
+
+    let local_addr = listener.local_addr().unwrap();
+    assert_eq!(local_addr.cid(), VMADDR_CID_HOST);
+    assert_eq!(local_addr.port(), LISTEN_PORT);
+}
+
+#[test]
+fn test_stream_addresses() {
+    let stream =
+        VsockStream::connect(&VsockAddr::new(SERVER_CID, SERVER_PORT)).expect("connection failed");
+
+    let local_addr = stream.local_addr().unwrap();
+    assert_eq!(local_addr.cid(), libc::VMADDR_CID_ANY);
```

Comment:
Got error: 
```
---- test_stream_addresses stdout ----
thread 'test_stream_addresses' panicked at 'assertion failed: `(left == right)`
  left: `2`,
 right: `4294967295`', tests/vsock.rs:95:5
```

Reply:
Weird, it passes for me. What OS/version/architecture are you testing on?

In any case, I've made the test accept either result.

Reply:
It works now.